### ISSUE TITLE
sublime plugin doesn't use html-tidy5

### ISSUE THREAD

**Issue by mark-rodgers** [SublimeLinter/SublimeLinter-html-tidy#44]
Cannot get these warnings on `<meta charset="UTF-8">` to go away.

tidy comes pre-installed on osx so I did this:

```
cp /usr/bin/tidy /usr/local/bin/tidy_old
brew install tidy-html5
```

Then I added `/usr/local/Cellar/tidy-html5/5.2.0/bin/` to my environmental variables.

Everything works correctly in my terminal (tidy is for html5) and (tidy_old is the older version), but for some reason it looks like Sublime is still using the old version of tidy because in tidy5 I should not be getting these meta errors anymore.. anyone have any suggestions?

![image](https://cloud.githubusercontent.com/assets/2798202/18650753/9d68bdd2-7e7b-11e6-8c6b-499d45e02006.png)

**tidy -v**

```
HTML Tidy for Mac OS X version 5.2.0
```

**tidy_old -v**

```
HTML Tidy for Mac OS X released on 31 October 2006 - Apple Inc. build 15.17
```

**tidy test.html**

```
Info: Document content looks like HTML5
No warnings or errors were found.

<!DOCTYPE html>
<html lang="en">
<head>
<meta name="generator" content=
"HTML Tidy for HTML5 for Mac OS X version 5.2.0">
<meta charset="UTF-8">
<title>Document</title>
</head>
<body>

</body>
</html>

About HTML Tidy: https://github.com/htacg/tidy-html5
Bug reports and comments: https://github.com/htacg/tidy-html5/issues
Official mailing list: https://lists.w3.org/Archives/Public/public-htacg/
Latest HTML specification: http://dev.w3.org/html5/spec-author-view/
Validate your HTML documents: http://validator.w3.org/nu/
Lobby your company to join the W3C: http://www.w3.org/Consortium

Do you speak a language other than English, or a different variant of
English? Consider helping us to localize HTML Tidy. For details please see
https://github.com/htacg/tidy-html5/blob/master/README/LOCALIZE.md
```

**tidy_old test.html**

```
line 4 column 5 - Warning: <meta> proprietary attribute "charset"
line 4 column 5 - Warning: <meta> lacks "content" attribute
Info: Document content looks like HTML Proprietary
2 warnings, 0 errors were found!

<!DOCTYPE html>
<html lang="en">
<head>
<meta name="generator" content=
"HTML Tidy for Mac OS X (vers 31 October 2006 - Apple Inc. build 15.17), see www.w3.org">
<meta charset="UTF-8">
<title>Document</title>
</head>
<body>

</body>
</html>

To learn more about HTML Tidy see http://tidy.sourceforge.net
Please send bug reports to html-tidy@w3.org
HTML and CSS specifications are available from http://www.w3.org/
Lobby your company to join W3C, see http://www.w3.org/Consortium
```

**Comment by lowgos** [comment]
I have exactly the same problem :/

**Comment by alexbet** [comment]
Same here.

**Comment by braver** [comment]
Sublime Text may not pick up on your PATH early enough and SublimeLinter 3 has a lot of problems. This should work better in [Sublime Linter 4](https://github.com/SublimeLinter/SublimeLinter#sublimelinter-4-is-imminent), you could try that.

**Comment by braver** [comment]
I cannot reproduce this problem. `brew install tidy-html5` replaces the already installed `tidy` without the need for any renaming, moving or editing PATH. `<meta charset="UTF-8">` is correctly recognised as valid HTML. This is on the latest SL4, which may help and is recommended anyway.

**Comment by kapiticoder** [comment]
I Also have this problem. 
I Followed  mark-rodgers suggestions, but get different results.

tidy -v
`HTML Tidy for Mac OS X released on 31 October 2006 - Apple Inc. build 15.18.1`

tidy_old -v
`HTML Tidy for Mac OS X released on 31 October 2006 - Apple Inc. build 15.18.1`

But
/usr/local/Cellar/tidy-html5/5.6.0/bin/tidy -v
`HTML Tidy for Apple macOS version 5.6.0`

This seems to be a paths problem but I have added 
`/usr/local/Cellar/tidy-html5/5.6.0/bin/`

To my `etc/paths` file.

The `etc/paths` file already has this path in it.
`/usr/local/bin`
Which is where an alias called "tidy" is located which points to 
`/usr/local/Cellar/tidy-html5/5.6.0/bin/tidy`

I have tried installing `Fix Mac Path` via package control into Sublime with No change.

I have researching this problem for hours, and I see that lots of people keep having this problem. Have I made a newbie mistake with my paths which would be sorted if the documentation included how to follow this instruction in the *SublimeLinter-html-tidy* readme.md file
```In order for `tidy` to be executed by SublimeLinter, you must ensure that its path is available to SublimeLinter.``` 

I am at a loss to know what to try next. The HTML 4 Linter works in Sublime, with SL4, but it keeps reporting tags like `<header>` as invalid.

**Comment by kapiticoder** [comment]
Here is my debug file
```
startup, version: 3200 osx x64 channel: stable
executable: /Applications/Sublime Text.app/Contents/MacOS/Sublime Text
working dir: /
packages path: /Users/wake003/Library/Application Support/Sublime Text 3/Packages
state path: /Users/wake003/Library/Application Support/Sublime Text 3/Local
zip path: /Applications/Sublime Text.app/Contents/MacOS/Packages
zip path: /Users/wake003/Library/Application Support/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
Unable to decode Packages/Edge Theme/assets/icons/night-skyscrollbar.png
pre session restore time: 0.326235
using gpu buffer for window
font face "Hack" could not be found, defaulting to "Menlo"
using gpu buffer for window
using gpu buffer for window
startup time: 0.462516
environment variables loaded using: /bin/bash -l
reloading plugin Default.arithmetic
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.colors
reloading plugin Default.comment
reloading plugin Default.convert_color_scheme
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.detect_indentation
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.rename
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin ShellScript.ShellScript
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.55-jsonschema
reloading plugin 0_package_control_loader.55-mdpopups
reloading plugin A File Icon.plugin
reloading plugin Edge Theme.E
reloading plugin Edge Theme.Icons
reloading plugin Fix Mac Path.FixPath
reloading plugin Git.git_commands
reloading plugin HyperlinkHelper.link_to_wikipedia_page_for_selection
reloading plugin HyperlinkHelper.lookup_with_google_and_link
reloading plugin HyperlinkHelper.wrap_selection_as_link
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin ReadmePlease.ReadmePlease
reloading plugin Sass.sass_completions
reloading plugin SideBarEnhancements.SideBar
reloading plugin SideBarEnhancements.SideBarAPI
reloading plugin SideBarEnhancements.SideBarDefaultDisable
reloading plugin SublimeLinter-csslint.linter
reloading plugin SublimeLinter-html-tidy.linter
htmltidy: Defining 'cls.syntax' has been deprecated. Use http://www.sublimelinter.com/en/stable/linter_settings.html#selector
reloading plugin SublimeLinter-php.linter
reloading plugin SublimeLinter.__init__
reloading plugin SublimeLinter.active_linters_view
reloading plugin SublimeLinter.busy_indicator_view
reloading plugin SublimeLinter.goto_commands
reloading plugin SublimeLinter.highlight_view
reloading plugin SublimeLinter.log_handler
reloading plugin SublimeLinter.message_view
reloading plugin SublimeLinter.panel_view
reloading plugin SublimeLinter.status_bar_view
reloading plugin SublimeLinter.sublime_linter
reloading plugin View In Browser.ViewInBrowserCommand
reloading plugin Browser Sync.browser_sync
reloading plugin ColorPicker.sublimecp
reloading plugin SFTP.SFTP
plugins loaded
Package Control: No updated packages
```

**Comment by braver** [comment]
@kapiticoder you don’t need to worry about Sublime Text or SublimeLinter just yet, you need to start at the beginning: setting up your PATH correctly on your Mac. You don’t need the PATH to be fixed for Sublime, it’s just not set up correctly for anything because `tidy` doesn’t point to the correct program. You already noticed this, so don’t continue to test and debug SublimeLinter but fix it first.
You say you edited `/etc/paths` which sounds a bit unorthodox to me, anything in etc is usually the OS’s playground, you don’t touch things in there unless you have to. I do notice StackOverflow pointing to this solution, but not everyone on SO knows what they’re doing. This is more like it, although I would put my stuff in .profile because it’s not necessarily specific for bash: https://coolestguidesontheplanet.com/add-shell-path-osx/.
So, continue trying to figure out how to edit PATH until `tidy -v` reports the required version. Then SublimeLinter will follow.

**Comment by kapiticoder** [comment]
Hi Braver,  Thanks very much for the link RE; Paths, it turns out the step I needed to do was to use the part on the page "Rearranging the default $PATH" to rearrange my "/use/local/bin" to the beginning of $PATH to take precedence. After a restart I got the correct response from tidy, and the Linter worked straight away in Sublime. I am reporting here to say thanks and also so that if anyone else finds this issue they can know how I solved it.
Thanks.
